virtwho/configure.py

```diff
@@ -1,4 +1,134 @@
-"""
-Used to manage virt-who configure file
-Add/Delete/Update options for /etc/virt-who.conf, /etc/virt-who.d/xxx.conf
-"""
+import os
+from virtwho.settings import Configure
+from virtwho.settings import config
+from virtwho.settings import TEMP_DIR
+from virtwho.ssh import SSHConnect
+from virtwho.logger import getLogger
+
+logger = getLogger(__name__)
+
+
+class VirtwhoHypervisorConfig:
+    """Able to create and manage /etc/virt-who.d/xxx.conf file when
+    call this class"""
+
+    def __init__(self, mode='esx', register_type='rhsm'):
+        """Create virt-who configuration file with basic options. All
+        data come from virtwho.ini. All local files are backed up
+        to /temp directory of the project root.
+        :param mode: The hypervisor mode.
+            (esx, xen, hyperv, rhevm, libvirt, kubevirt, local)
+        :param register_type: The subscription server. (rhsm, satellite)
+        """
+        self.register = get_register_handler(register_type)
+        self.hypervisor = get_hypervisor_handler(mode)
+        self.remote_ssh = virtwho_ssh_connect(mode)
+        self.local_file = os.path.join(TEMP_DIR, f'{mode}.conf')
+        self.remote_file = f'/etc/virt-who.d/{mode}.conf'
+        self.cfg = Configure(self.local_file, self.remote_ssh, self.remote_file)
+
+        self.section = f'virtwho-{mode}'
+        if mode is 'local':
+            self.update('type', 'libvirt')
+        else:
+            self.update('type', mode)
```

Comment:
`self.update('type', mode)` is used for all the hypervisors, it's not necessary to define two lines, but...it's up to you

Reply:
If not define ```self.update('type', 'libvirt')``` for ```local``` mode, ```type=local``` will be used for local.
But will find more smart way.

Reply:
Now I got your point, it's true :+1: